crates/sui-types/src/event.rs

```diff
@@ -22,19 +23,27 @@ use crate::{
 #[derive(Debug, Clone, PartialEq)]
 pub struct EventEnvelope {
     /// UTC timestamp in milliseconds since epoch (1/1/1970)
-    timestamp: u64,
+    timestamp: u128,
     /// Transaction digest of associated transaction, if any
     tx_digest: Option<TransactionDigest>,
     /// Specific event type
-    event: Event,
+    pub event: Event,
+    /// JSON contents for MoveEvent, if present
+    pub move_event_json_contents: Option<Value>,
```

Comment:
resolved offline.

regarding `OnceCell`, I wanted to use it but it would be a bit ugly because the necessary `ModuleResolver` is not part of the struct. I will leave it as it is